compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrClassifierStorage.kt

```diff
@@ -380,10 +380,11 @@ class Fir2IrClassifierStorage(
         require(index >= 0)
         val origin = typeParameter.computeIrOrigin()
         val irTypeParameter = with(typeParameter) {
+            val replacedSelfName = if (name.asString() == "<Self>") Name.identifier("\$Self") else name
```

Comment:
Please use predefined constant instead of `Name.identifier("\$Self")`

Reply:
Added `SELF_TYPE_IDENTIFIER` to companion object.